src/util/glTF/glTFContainer.js

```diff
@@ -127,43 +128,91 @@ x3dom.glTF.glTFTexture.prototype.isPowerOfTwo = function(x)
     return powerOfTwo;
 };
 
+x3dom.glTF.glTFTexture.prototype.needsPowerOfTwo = function(gl)
+{
+    var resize = true;
+    resize &= (this.sampler.magFilter == gl.LINEAR || this.sampler.magFilter == gl.NEAREST);
+    resize &= (this.sampler.minFilter == gl.LINEAR || this.sampler.minFilter == gl.NEAREST);
+    resize &= (this.sampler.wrapS == gl.CLAMP_TO_EDGE);
+    resize &= (this.sampler.wrapT == gl.CLAMP_TO_EDGE);
+
+    return !resize;
+};
+
+x3dom.glTF.glTFTexture.prototype.needsMipMaps = function(gl)
+{
+    var need = true;
+    need &= (this.sampler.magFilter == gl.LINEAR || this.sampler.magFilter == gl.NEAREST);
+    need &= (this.sampler.minFilter == gl.LINEAR || this.sampler.minFilter == gl.NEAREST);
+
+    return !need;
+};
+
+x3dom.glTF.glTFTexture.prototype.nextPowerOfTwo = function(n)
+{
+    return Math.pow(2, Math.ceil(Math.log(n)/Math.log(2)));
+};
+
 x3dom.glTF.glTFTexture.prototype.create = function(gl)
 {
     if(this.image.complete == false)
         return;
 
     this.glTexture = gl.createTexture();
 
+    var imgSrc = this.image;
+
+    if(this.needsPowerOfTwo(gl)){
+        var width = this.image.width;
+        var height = this.image.height;
+
+        var aspect = width / height;
+
+        var width2 = this.nextPowerOfTwo(width);
+        var height2 = this.nextPowerOfTwo(height);
+
+        var aspect2 = width2 / height2;
+
+        if(aspect - aspect2 > 0.01){
+            console.warn("Image "+this.image.src+" needs to be resized to power of two, but has unsupported aspect ratio and may be distorted!");
+        }
+
+        var oc = document.createElement('canvas'),
```

Comment:
At this point you can also use the Utils.scaleImage function
https://github.com/x3dom/x3dom/blob/master/src/util/Utils.js#L634